mmv1/third_party/terraform/services/bigquery/resource_bigquery_table_test.go

```diff
@@ -2944,6 +3031,196 @@ resource "google_bigquery_table" "test" {
 `, datasetID, tableID)
 }
 
+func testAccBigQueryTablePrimaryKey(datasetID, tableID string) string {
+	return fmt.Sprintf(`
+  resource "google_bigquery_dataset" "foo" {
+    dataset_id = "%s"
+  }
+
+  resource "google_bigquery_table" "test" {
+	  deletion_protection = false
+    table_id   = "%s"
+    dataset_id = google_bigquery_dataset.foo.dataset_id
+
+    table_constraints {
+		primary_key {
+			columns = ["id"]
+		}
+    }
+
+    schema = <<EOH
+	[
+		{
+			"name": "ts",
+			"type": "TIMESTAMP"
+		},
+		{
+			"name": "id",
+			"type": "INTEGER"
+		}
+	]
+	EOH
+	}
+  `, datasetID, tableID)
+}
+
+func testAccBigQueryTableForeignKeys(projectID, datasetID, tableID_pk, tableID_fk string) string {
+	return fmt.Sprintf(`
+  resource "google_bigquery_dataset" "foo" {
+    dataset_id = "%s"
+  }
+
+  resource "google_bigquery_table" "table_pk" {
```

Comment:
What is the reason to add resource `google_bigquery_table..table_pk`? It looks like resource `google_bigquery_table.test` is used as the testing resource by specifying `resourceName := "google_bigquery_table.test"`.

Reply:
@zli82016 A foreign key in a table needs to be linked to a primary key in another table. This is currently not enforced by BigQuery, however I added a primary key table anyway because this is the way it's intended to be used by users, and in case these constraints become enforced.